### ISSUE TITLE
Exception when analyzing - Can not find source for .class

### ISSUE THREAD

**Issue by florianli** [sdaschner/jaxrs-analyzer-maven-plugin#3]
Hello,

while analyzing my project, the build fails because of a RuntimeExcetion

the debug output is (Some names are made unrecognizable with X)
[INFO] --- jaxrs-analyzer-maven-plugin:0.5:analyze-jaxrs (default) @ RMSDatenhaltung ---

[DEBUG] Configuring mojo com.sebastian-daschner:jaxrs-analyzer-maven-plugin:0.5:analyze-jaxrs from plugin realm ClassRealm[plugin>com.sebastian-daschner:jaxrs-analyzer-maven-plugin:0.5, parent: sun.misc.Launcher$AppClassLoader@4e25154f]
[DEBUG] Configuring mojo 'com.sebastian-daschner:jaxrs-analyzer-maven-plugin:0.5:analyze-jaxrs' with basic configurator -->
[DEBUG](f) backend = plaintext
[DEBUG](f) buildDirectory = C:\data\EclipseWorkspace\RMSDatenhaltung\target
[DEBUG](f) deployedDomain = example.com
[DEBUG](f) outputDirectory = C:\data\EclipseWorkspace\RMSDatenhaltung\target\classes
[DEBUG](f) project = MavenProject: XXX.XXX.XXX.XXX:RMSDatenhaltung:2.0.0-SNAPSHOT @ C:\data\EclipseWorkspace\RMSDatenhaltung\pom.xml
[DEBUG] -- end configuration --
[INFO] analyzing JAX-RS resources, using Plain text backend
[DEBUG] Dependency paths are: [C:\Maven\m2\repository\XXX\XXX\XXX\XXX\RMSSchema\2.3\RMSSchema-2.3.jar, C:\Maven\m2\repository\org\jboss\spec\javax\ejb\jboss-ejb-api_3.1_spec\1.0.2.Final-redhat-2\jboss-ejb-api_3.1_spec-1.0.2.Final-redhat-2.jar, C:\Maven\m2\repository\org\jboss\shrinkwrap\resolver\shrinkwrap-resolver-depchain\2.0.0\shrinkwrap-resolver-depchain-2.0.0.pom, C:\Maven\m2\repository\org\infinispan\infinispan-core\7.0.3.Final\infinispan-core-7.0.3.Final.jar, C:\Maven\m2\repository\org\jboss\spec\javax\annotation\jboss-annotations-api_1.1_spec\1.0.1.Final-redhat-2\jboss-annotations-api_1.1_spec-1.0.1.Final-redhat-2.jar, C:\Maven\m2\repository\org\jboss\arquillian\protocol\arquillian-protocol-servlet\1.1.0.Final\arquillian-protocol-servlet-1.1.0.Final.jar, C:\Maven\m2\repository\org\apache\maven\maven-artifact\3.2.3\maven-artifact-3.2.3.jar, C:\Maven\m2\repository\XX\XXX\XXXXX\0.3.4.2\XXXX-0.3.4.2.jar, C:\Maven\m2\repository\javax\enterprise\cdi-api\1.0-SP4\cdi-api-1.0-SP4.jar, C:\Maven\m2\repository\org\jboss\resteasy\resteasy-jaxb-provider\2.3.8.Final-redhat-3\resteasy-jaxb-provider-2.3.8.Final-redhat-3.jar, C:\Maven\m2\repository\com\fasterxml\jackson\core\jackson-core\2.2.3\jackson-core-2.2.3.jar, C:\Maven\m2\repository\com\google\guava\guava\13.0.1-redhat-1\guava-13.0.1-redhat-1.jar, C:\Maven\m2\repository\org\jboss\arquillian\junit\arquillian-junit-container\1.1.0.Final\arquillian-junit-container-1.1.0.Final.jar, C:\Maven\m2\repository\org\apache\deltaspike\core\deltaspike-core-api\1.2.1\deltaspike-core-api-1.2.1.jar, C:\Maven\m2\repository\org\apache\deltaspike\core\deltaspike-core-impl\1.2.1\deltaspike-core-impl-1.2.1.jar, C:\Maven\m2\repository\com\sun\xml\txw2\txw2\20110809-redhat-4\txw2-20110809-redhat-4.jar, C:\Maven\m2\repository\org\mockito\mockito-core\1.9.5\mockito-core-1.9.5.jar, C:\Maven\m2\repository\commons-logging\commons-logging\1.2\commons-logging-1.2.jar, C:\Maven\m2\repository\org\infinispan\infinispan-cachestore-leveldb\7.0.3.Final\infinispan-cachestore-leveldb-7.0.3.Final.jar, C:\Maven\m2\repository\org\jboss\resteasy\resteasy-jaxrs\2.3.8.Final-redhat-3\resteasy-jaxrs-2.3.8.Final-redhat-3.jar, C:\Maven\m2\repository\org\jboss\resteasy\resteasy-multipart-provider\2.3.8.Final-redhat-3\resteasy-multipart-provider-2.3.8.Final-redhat-3.jar, C:\Maven\m2\repository\com\fasterxml\jackson\core\jackson-annotations\2.2.3\jackson-annotations-2.2.3.jar, C:\Maven\m2\repository\com\fasterxml\jackson\core\jackson-databind\2.2.3\jackson-databind-2.2.3.jar, C:\Maven\m2\repository\org\jboss\spec\javax\ws\rs\jboss-jaxrs-api_1.1_spec\1.0.1.Final-redhat-2\jboss-jaxrs-api_1.1_spec-1.0.1.Final-redhat-2.jar, C:\Maven\m2\repository\org\jboss\spec\javax\servlet\jboss-servlet-api_3.0_spec\1.0.2.Final-redhat-1\jboss-servlet-api_3.0_spec-1.0.2.Final-redhat-1.jar, C:\Maven\m2\repository\org\jboss\jboss-vfs\3.2.5.Final-redhat-1\jboss-vfs-3.2.5.Final-redhat-1.jar, C:\Maven\m2\repository\junit\junit\4.11\junit-4.11.jar, C:\Maven\m2\repository\org\jboss\logging\jboss-logging\3.1.2.GA\jboss-logging-3.1.2.GA.jar, C:\Maven\m2\repository\com\orientechnologies\orientdb-core\2.0.10\orientdb-core-2.0.10.jar]
[DEBUG] Project paths are: [C:\data\EclipseWorkspace\RMSDatenhaltung\target\classes]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 13.034 s
[INFO] Finished at: 2015-06-18T14:45:10+01:00
[INFO] Final Memory: 49M/378M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.sebastian-daschner:jaxrs-analyzer-maven-plugin:0.5:analyze-jaxrs (default) on project RMSDatenhaltung: Execution default of goal com.sebastian-daschner:jaxrs-analyzer-maven-plugin:0.5:analyze-jaxrs failed: cannot find XXX\XXX\XXX\datenhaltung\storage\db\util\ODocToModelConverter: XXX.XXX.XXX.datenhaltung.storage.db.util.ODocToModelConverter found in XXX\XXX\XXX\datenhaltung\storage\db\util\ODocToModelConverter.class -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.sebastian-daschner:jaxrs-analyzer-maven-plugin:0.5:analyze-jaxrs (default) on project RMSDatenhaltung: Execution default of goal com.sebastian-daschner:jaxrs-analyzer-maven-plugin:0.5:analyze-jaxrs failed: cannot find XXX\XXX\XXX\datenhaltung\storage\db\util\ODocToModelConverter: XXX.XXX.XXX.datenhaltung.storage.db.util.ODocToModelConverter found in XXX\XXX\XXX\datenhaltung\storage\db\util\ODocToModelConverter.class
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:224)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:108)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:76)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:116)
    at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:361)
    at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:155)
    at org.apache.maven.cli.MavenCli.execute(MavenCli.java:584)
    at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:213)
    at org.apache.maven.cli.MavenCli.main(MavenCli.java:157)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:497)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default of goal com.sebastian-daschner:jaxrs-analyzer-maven-plugin:0.5:analyze-jaxrs failed: cannot find XXX\XXX\XXX\datenhaltung\storage\db\util\ODocToModelConverter: XXX.XXX.XXX.datenhaltung.storage.db.util.ODocToModelConverter found in XXX\XXX\XXX\datenhaltung\storage\db\util\ODocToModelConverter.class
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:144)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
    ... 19 more
Caused by: java.lang.RuntimeException: cannot find XXX\XXX\XXX\datenhaltung\storage\db\util\ODocToModelConverter: XXX.XXX.XXX.datenhaltung.storage.db.util.ODocToModelConverter found in XXX\XXX\XXX\datenhaltung\storage\db\util\ODocToModelConverter.class
    at javassist.CtClassType.getClassFile2(CtClassType.java:194)
    at javassist.CtClassType.getAnnotations(CtClassType.java:541)
    at javassist.CtClassType.getAvailableAnnotations(CtClassType.java:531)
    at com.sebastian_daschner.jaxrs_analyzer.analysis.project.classes.ClassAnalyzer.isRelevant(ClassAnalyzer.java:131)
    at com.sebastian_daschner.jaxrs_analyzer.analysis.project.classes.ClassAnalyzer.analyze(ClassAnalyzer.java:57)
    at com.sebastian_daschner.jaxrs_analyzer.analysis.ProjectAnalyzer$$Lambda$18/2065199032.apply(Unknown Source)
    at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
    at java.util.HashMap$KeySpliterator.forEachRemaining(HashMap.java:1540)
    at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:512)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:502)
    at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
    at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
    at com.sebastian_daschner.jaxrs_analyzer.analysis.ProjectAnalyzer.analyze(ProjectAnalyzer.java:78)
    at com.sebastian_daschner.jaxrs_analyzer.JAXRSAnalyzer.analyze(JAXRSAnalyzer.java:69)
    at com.sebastian_daschner.jaxrs_analyzer.maven.JAXRSAnalyzerMojo.execute(JAXRSAnalyzerMojo.java:117)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:133)
    ... 20 more

**Comment by sdaschner** [comment]
Hi,

Is the `ODocToModelConverter` included in your project (RMSDatenhaltung)?
Could you please make an example how that annotation is used in your classes (you can make a generic example like class A has annotation XY, etc.).

**Comment by florianli** [comment]
Hello,

thank you for your quick answer. Sorry that it took a while for my reply.
The `ODocToModelConverter` is a simple java class with a bunch of static methods. It is not an annotation.

**Comment by sdaschner** [comment]
Hi,

Finally I managed to reproduce and fix this error (was a Windows path issue, actually). This is already included in the latest snapshot version `0.6-SNAPSHOT`. The Maven plugin snapshot repository has to be added for that:

```
<pluginRepositories>
    <pluginRepository>
        <id>ossrh-snapshots</id>
        <url>https://oss.sonatype.org/content/repositories/snapshots/</url>
        <releases>
            <enabled>false</enabled>
        </releases>
        <snapshots>
            <enabled>true</enabled>
        </snapshots>
    </pluginRepository>
</pluginRepositories>
```

Anyway, I'll publish the `0.6` version later this weekend.

**Comment by sdaschner** [comment]
FYI: I've just released version `0.6`.
